apps/dav/lib/BackgroundJob/UserStatusAutomation.php

```diff
@@ -139,6 +139,12 @@ protected function run($argument) {
 			}
 		}
 
+		if (empty($nextPotentialToggles)) {
+			$this->logger->info('Removing ' . self::class . ' background job for user "' . $userId . '" because the user has no valid availability rules set');
+			$this->jobList->remove(self::class, $argument);
```

Comment:
```suggestion
			$this->jobList->remove(self::class, $argument);
			$this->manager->revertUserStatus($userId, IUserStatus::MESSAGE_AVAILABILITY, IUserStatus::DND);
```

I would also remove the status if it was added automatically

Reply:
Pushed a matching unit test to #36852